lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -207,6 +209,64 @@ spec = do
             [ expectFieldEqual delegation (Delegating (p2 ^. #id))
             ]
 
+    describe "STAKE_POOLS_JOIN_01x - Fee boundary values" $ do
+
+        it "STAKE_POOLS_JOIN_01x - \
+            \I can join if I have anything to delegate" $ \(ctx) -> do
+            (_, p:_) <- eventually $
+                unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+            let fee = 42
+            w <- fixtureWalletWith ctx [fee + 1]
+            joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")>>= flip verify
+                [ expectResponseCode HTTP.status200
+                , expectFieldEqual status Pending
+                , expectFieldEqual direction Outgoing
+                ]
+
+        it "STAKE_POOLS_JOIN_01x - \
+            \I cannot join when I have enough fee but 0 balance to delegate" $ \ctx -> do
+            (_, p:_) <- eventually $
+                unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+            let fee = 42
+            w <- fixtureWalletWith ctx [fee]
+            r <- joinStakePool ctx (p ^. #id) (w, "Secure Passphrase")
+            print r
+            expectResponseCode HTTP.status403 r
+            expectErrorMessage errMsg403Fee r
+            -- the error msg here is a bit misleading:
+            -- I'm unable to adjust the given transaction to cover the associated fee!
+            -- In order to do so, I'd have to select one or more additional inputs,
+            -- but I can't do that without increasing the size of the transaction
+            -- beyond the acceptable limit. Note that I am only missing 42 Lovelace.
```

Comment:
Hmmm. That's an interesting case which could deserve some more investigation.

Reply:
My understanding was that the behavior is correct, cause it doesn't make sense to delegate 0 funds. In that case adjusting error message should be enough... (?)

Reply:
It's not really about delegating 0 funds. When you create the delegation certificates, it's not like taking a snapshot of your current funds and delegating them. It's simply creates a note that tells the node that any address containing your reward account (grouped address) should be delegated to a particular pool. So, all addresses, before and after the certificate submission carrying a ref to the reward account will be delegated. 

There's no problem to create a certificate and have no funds to delegate _yet_.